src/OrbitBase/File.cpp

```diff
@@ -61,4 +72,22 @@ ErrorMessageOr<void> WriteFully(const unique_fd& fd, std::string_view content) {
   return outcome::success();
 }
 
+ErrorMessageOr<size_t> ReadFully(const unique_fd& fd, void* buffer, size_t size) {
```

Comment:
I guess that this depends on what use you have in mind for this function, but it's annoying to have to provide a preallocated buffer and to have to guess the size. It makes for a more complicated replacement for `read` while basically providing the same interface. In particular, in doesn't provide the convenience that one would expect from `Fully`.
Can't a vector of bytes be returned, maybe passing a *maximum size* instead of the size of the buffer?

Reply:
There are couple of reason for doing it this way:
1. Usually the pattern in which this is used is when we repeatedly read chunks from file and do some computations.
2. The only(!) case when actual is less then requested size is end of file.
3. There are 2 other functions that let you read whole file and (coming soon) the one that reads one line, they return std::string

The convenience is still there since you do not need to handle interruptions by signals (see 2).

Reply:
Sure, but then it shouldn't be called `ReadFully`, all your three points are basically stating that "fully" is the opposite of what this function is doing.